src/machine/machine_mimxrt1062.go

```diff
@@ -879,3 +880,190 @@ func (p Pin) getMuxMode(config PinConfig) uint32 {
 		panic("machine: invalid pin mode")
 	}
 }
+
+type ADCConfig struct {
+	Resolution uint32
+	Samples    uint32
+}
+
+// InitADC is not used by this machine. Use `(ADC).Configure()`.
+func InitADC() {}
+
+// Configure initializes the receiver ADC's Pin and the ADC1/ADC2 instances for
+// immediate usage with default settings.
+func (a ADC) Configure() { a.ConfigureMode(ADCConfig{}) }
+
+// ConfigureMode initializes the receiver ADC's Pin and the ADC1/ADC2 instances
+// for immediate usage with given settings.
+func (a ADC) ConfigureMode(config ADCConfig) {
+
+	const (
+		defaultResolution = uint32(10)
+		defaultSamples    = uint32(4)
+	)
+
+	a.Pin.Configure(PinConfig{Mode: PinInputAnalog})
+
+	resolution, samples := config.Resolution, config.Samples
+	if 0 == resolution {
+		resolution = defaultResolution
+	}
+	if 0 == samples {
+		samples = defaultSamples
+	}
+
+	mode, average := a.mode(resolution, samples)
+
+	nxp.ADC1.CFG.Set(mode | nxp.ADC_CFG_ADHSC) // configure ADC1
+	nxp.ADC2.CFG.Set(mode | nxp.ADC_CFG_ADHSC) // configure ADC2
+
+	// begin calibration
+	nxp.ADC1.GC.Set(average | nxp.ADC_GC_CAL)
+	nxp.ADC2.GC.Set(average | nxp.ADC_GC_CAL)
+
+	adcStatusGlobal.startCalibrating()
+	adcStatusGlobal.waitForCalibration()
+}
+
+func (a ADC) Get() uint16 {
```

Comment:
Hmm, it seems I have unintentionally lost a commit (during a previous rebase, presumably). 

However, I was actually _scaling_ the returned value uniformly to 16 bits [0, 65535], instead of just left-shifting.

For the 12-bit case, I don't think it matters much, as you'll just have a slight jump from 0 to 16 in the outputs, and a uniform distribution thereafter to 65535.

But for the 8-bit case, left-shifting seems like a bad idea, with a gap from 0 to 256. That seems more noticeable.

I don't believe performance between the two alternatives matters here, with scaling requiring just 1 integer multiply + 1 integer divide (if I remember correctly). The real bottleneck in ADC reads will be the conversions themselves. For example, with default settings (10-bit (requires 20 ADC clock cycles), 4 samples), each ADC read will require 80 ADC clock cycles just to perform the conversion.

I will add the docs to this function along with one of these 2 scaling techniques. Is it okay if I do the arithmetic scaling (not the bit-shifting)?

Reply:
Added godocs and 16-bit scaling to method `Get`